lib/IRGen/IRGen.cpp

```diff
@@ -156,6 +156,13 @@ swift::getIRTargetOptions(IRGenOptions &Opts, ASTContext &Ctx) {
 
   auto *Clang = static_cast<ClangImporter *>(Ctx.getClangModuleLoader());
   clang::TargetOptions &ClangOpts = Clang->getTargetInfo().getTargetOpts();
+
+  // WebAssembly doesn't support atomics or DWARF5 yet.
+  if (Clang->getTargetInfo().getTriple().isOSBinFormatWasm()) {
+    TargetOpts.DebuggerTuning = llvm::DebuggerKind::Default;
```

Comment:
Why change the debugger tuning?  That controls whether the debug info is better suited for LLDB or for GDB.  You want the DebugVersion.

Reply:
Tuning for LLDB causes llvm to always generate DWARF5 accel tables: https://github.com/apple/swift-llvm/blob/8d1b92b83c0bd9735f4b8771b26ce96f8c2a3724/lib/CodeGen/AsmPrinter/DwarfDebug.cpp#L300

I'll change the comment to clarify this.

Reply:
llvm has a -accel-tables option, but that isn't a great solution either. I think the best solution would be to lower the DWARF version to DWARF4 for the WASM platform.

Reply:
DWARF version is already set to 4 here https://github.com/apple/swift/blob/5bef9f21187d4297253892a2220737ef222d2689/include/swift/Basic/Dwarf.h#L24, and we don't set it to 5 anywhere for Wasm, so I assume these tweaks are still needed to turn off accel tables? Or would you suggest us to use some other setting for this?